src/button/index.tsx

```diff
@@ -189,7 +189,7 @@ export default class Button extends PureComponent<ButtonProps, ButtonState> {
             onDoneAnimationEnd={onDoneAnimationEnd}
           />
         )}
-        {children}
+        {!hasLoader && children}
```

Comment:
hasLoader does the job for LOADING & CHECKED ?

Reply:
Add unit tests

Reply:
Yep Done

Reply:
And should have a discussion about the collapsed display. I agree hasLoader is a bit more than just managing loading state.